vision_model/trainer_callback_interface.py

```diff
@@ -0,0 +1,51 @@
+import abc
+import typing
+import torch
+
+
+class TrainerCallbackInterface(abc.ABC):
+    """Interfaces a Trainer provides for callback methods"""
+    @property
+    @abc.abstractmethod
+    def train_dataloader(self) -> torch.utils.data.DataLoader:
+        raise NotImplementedError
+
+    @property
+    @abc.abstractmethod
+    def val_dataloader(self) -> typing.Optional[torch.utils.data.DataLoader]:
+        raise NotImplementedError
+
+    @property
+    @abc.abstractmethod
+    def num_epochs(self) -> int:
+        raise NotImplementedError
+
+    @property
```

Comment:
many recent tasks like image-text matching or captioning do not has num_classes any more, should we still have it here?

Reply:
This is related to the question in the another comment above. Should we make interface for inference-only model?

Reply:
yeah, I think so and we are using/needing it anyway

Reply:
Inference-only interface has been created. Also removed num_classes since models should know about it at init().